src/kaleidoscope/plugin/LED-Wavepool.cpp

```diff
@@ -27,52 +27,64 @@ namespace plugin {
 #define MS_PER_FRAME 40  // 40 = 25 fps
 #define FRAMES_PER_DROP 120  // max time between raindrops during idle animation
 
-int8_t WavepoolEffect::surface[2][WP_WID * WP_HGT];
-uint8_t WavepoolEffect::page = 0;
-uint8_t WavepoolEffect::frames_since_event = 0;
 uint16_t WavepoolEffect::idle_timeout = 5000;  // 5 seconds
 int16_t WavepoolEffect::ripple_hue = WavepoolEffect::rainbow_hue; // automatic hue
 
 // map native keyboard coordinates (16x4) into geometric space (14x5)
-PROGMEM const uint8_t WavepoolEffect::rc2pos[ROWS * COLS] = {
+PROGMEM const uint8_t WavepoolEffect::DynamicLEDMode::rc2pos[ROWS * COLS] = {
   0,  1,  2,  3,  4,  5,  6,     59, 66,    7,  8,  9, 10, 11, 12, 13,
   14, 15, 16, 17, 18, 19, 34,    60, 65,   35, 22, 23, 24, 25, 26, 27,
   28, 29, 30, 31, 32, 33, 48,    61, 64,   49, 36, 37, 38, 39, 40, 41,
   42, 43, 44, 45, 46, 47,     58, 62, 63, 67,    50, 51, 52, 53, 54, 55,
 };
 
+WavepoolEffect::DynamicLEDMode::DynamicLEDMode(const WavepoolEffect *parent)
+  : parent_(parent),
+    frames_since_event_(0),
+    surface_{},
+    page_(0)
+{}
+
 EventHandlerResult WavepoolEffect::onKeyswitchEvent(Key &mapped_key, byte row, byte col, uint8_t key_state) {
   if (row >= ROWS || col >= COLS)
     return EventHandlerResult::OK;
 
+  if (::LEDControl.get_mode_index() != led_mode_id_)
+    return EventHandlerResult::OK;
+
+  return ::LEDControl.get_mode<DynamicLEDMode>()
+         ->onKeyswitchEvent(mapped_key, row, col, key_state);
+}
+
+EventHandlerResult WavepoolEffect::DynamicLEDMode::onKeyswitchEvent(Key &mapped_key, byte row, byte col, uint8_t key_state) {
   if (keyIsPressed(key_state)) {
     uint8_t offset = (row * COLS) + col;
-    surface[page][pgm_read_byte(rc2pos + offset)] = 0x7f;
-    frames_since_event = 0;
+    surface_[page_][pgm_read_byte(rc2pos + offset)] = 0x7f;
+    frames_since_event_ = 0;
```

Comment:
I know I'm usually the one asking for more conformity with the coding style guides, but these name changes are unrelated to the rest of this very large PR, which is difficult enough to decipher as it is.

Reply:
As (technically) I had to introduce new classes, I implemented them according to the rules of the style guide.

Reply:
If this is a new class, why is the diff showing only these variable names as having changed?

Reply:
I do see your point. However, these name changes make it even more difficult to figure out what the relevant changes are for this PR. It would be much clearer if you at least separated these name changes into a different commit, and even better if they were in a separate PR entirely, since they affect plenty of lines of code that are not related to the dynamic LED mode changes.